src/pivot-table/hooks/use-selections-model.ts

```diff
@@ -21,17 +19,57 @@ export interface SelectedPivotCell {
   qCol: number;
 }
 
+type SelectedRowOrColumn = {
+  selectionCellType: NxSelectionCellType | null;
+  coord: number;
+};
+
+const getNextState = (cell: Cell, selectedPivotCells: Set<Cell>, selectedRowOrColumn: SelectedRowOrColumn) => {
+  const nextSelectedPivotCells = new Set(selectedPivotCells);
+  const nextSelectedRowOrColumn = { ...selectedRowOrColumn };
+
+  if (nextSelectedPivotCells.has(cell)) {
+    nextSelectedPivotCells.delete(cell);
+
+    if (nextSelectedPivotCells.size === 0) {
+      nextSelectedRowOrColumn.selectionCellType = null;
+      nextSelectedRowOrColumn.coord = -1;
+    }
+
+    return {
+      nextSelectedPivotCells,
+      nextSelectedRowOrColumn,
+    };
+  }
+
+  nextSelectedPivotCells.add(cell);
+  nextSelectedRowOrColumn.selectionCellType = cell.selectionCellType;
+  nextSelectedRowOrColumn.coord = cell.selectionCellType === NxSelectionCellType.NX_CELL_LEFT ? cell.x : cell.y;
+
+  return {
+    nextSelectedPivotCells,
+    nextSelectedRowOrColumn,
+  };
+};
+
 export default function useSelectionsModel(
   selections: ExtendedSelections,
   updatePageInfo: (args: Partial<PageInfo>) => void,
 ): SelectionModel {
   const isActive = selections.isActive();
-  const [selected, setSelected] = useState<SelectedPivotCell[]>([]);
+  const [selectedPivotCells, setSelectedPivotCells] = useState<Set<Cell>>(new Set());
+  const [selectedRowOrColumn, setSelectedRowOrColumn] = useState<SelectedRowOrColumn>({
+    selectionCellType: null,
+    coord: -1,
+  });
```

Comment:
this value will only going to be point to the last cell that processed and keep itself updated by it's info, but within the isLocked function (which could receive **any** cell) we rely on (last processed) value to compute the locked state (line 89 - 102). 

(or maybe) am i miss-interpreting the changes?

Reply:
I think the answer in one of the other comments should answer this as well?